pandas/core/indexes/base.py

```diff
@@ -483,6 +483,26 @@ def _simple_new(cls, values, name: Label = None):
     def _constructor(self):
         return type(self)
 
+    def _maybe_check_unique(self):
+        if not self.is_unique:
+            # TODO: position, value, not too large.
+            msg = """Index has duplicates."""
+            duplicates = self._format_duplicate_message()
+            msg += "\n{}".format(duplicates)
+
+            raise DuplicateLabelError(msg)
+
+    def _format_duplicate_message(self):
+        from pandas import Series
+
+        duplicates = self[self.duplicated(keep="first")].unique()
```

Comment:
can you add a doc-string here. i see that you are forming the list of duplicates; this *could* potentially be very large, which i guess is ok?

Reply:
I think we're probably OK here. The frame we construct won't ever be any larger than the input.